lessons/git/git-collaboration-2in1/index.md

```diff
@@ -0,0 +1,291 @@
+kde vysvětlil, proč hodlá vytvořit operační systém s otevřeným kódem a odstartoval tak hnutí svobodného softwaru.
+To prosazuje 4 následujících svobody (převzato z [Wikipedie](https://cs.wikipedia.org/wiki/Svobodn%C3%BD_software)):
+
+1. svoboda používat program za jakýmkoliv účelem,
+2. svoboda studovat, jak program pracuje a možnost přizpůsobit ho svým potřebám,
+3. svoboda redistribuovat kopie programu,
+4. svoboda vylepšovat program a zveřejňovat zlepšení, aby z nich mohla mít prospěch celá komunita.
+ 
+Dnes je spousta projektů s otevřeným zdrojovým kódem dostupných na Internetu a každý je používáme. 
+Jejich další sdílení je upraveno jednou z licencí, které tyto základní svobody zaručují. 
+
+!!! note "Pozor"
+    Termíny **open-source** a **free software** nejsou zcela zaměnitelné, ale pro naše účely je zatím můžeme chápat jako synonyma.
+
+Spoustu open-source projektů najdete právě na GitHubu. Ne všechny jsou v Pythonu. Ne všechny jsou kvalitní –- 
+protože si každý může zveřejnit co chce, na GitHubu se válí spousta nedodělků, opuštěných nápadů a nepodařených experimentů. 
+A bohužel, ne všechny projekty mají přátelské autory.
+
+Na druhou stranu ale open-source programy mají svoje výhody: nejenom se z nich může kdokoli učit, 
+ale každý může i zkontrolovat, jestli dělají to co dělat mají. 
+Populární open-source programy nás například pravděpodobně nebudou špehovat (tj. hlásit autorovi, co na počítači děláme), 
+ani většinou neobsahují reklamy: kdyby to dělaly, najde se někdo kdo tyhle „funkce” odstraní 
+a lidé časem začnou používat opravenou verzi. 
+
+Některé příklady populárních open-source projektů:
+
+* Mozilla Firefox, Chromium (prohlížeče)
+* Atom, gedit (textové editory)
+* CPython (jazyk Python)
+* Linux, Android (jádra operačních systémů)
+* Pytest (pythoní knihovna na testování)
+* Django, Flask, Requests (webové knihovny pro Python)
+* Numpy, Jupyter, Matplotlib (pythoní knihovny pro vědce a analytiky)
+* Materiály k tomuto kurzu     
+
+A jak vidno z posledního příkladu, nejen softwarové projekty se dají vést takhle veřejně. Tento kurz vychází z principů open source: 
+všechno know-how je sdílené, a budeme rádi, když se zapojíš.
+
+Pokud vás tato problematika zajímá, doporučujeme krátkou knihu [Katedrála a tržiště](https://www.root.cz/knihy/katedrala-a-trziste/), která nabízí srovnání open-source s tradičním vývojem softwaru za zavřenými dveřmi.
+
+Pokud budete někdy začínat nový projekt a budete chtít, aby z něj měl prospěch někdo další,
+stojí za to uvolnit ho pod svobodnou licencí a dát třeba právě na GitHub.
+Licence projektu většinou najdete v souboru `LICENCE`. 
+Na kód neuvolněný pod svobodnou licencí se automaticky vztahuje copyright a ostatní ho nemohou volně sdílet.
+
+Informace o open-source licencích najdete např. na [choosealicence.com](https://choosealicense.com), případně [creativecommons.org](https://creativecommons.org) a [opensource.org](https://opensource.org).
+
+## Git a GitHub
```

Comment:
Neřeším větvení a konflikty, protože to holky zatím nikdy nepobrali. Udělají vlastně tu úlohu z lekce Spolupráce. Je to celé taková kompilace těch 2 lekcí + trochu něco ode mě.